lib/feature/auth/success_message/screen/success_message_screen.dart

```diff
@@ -0,0 +1,124 @@
+  const SuccessMessageScreenWithBloc({super.key});
+
+  @override
+  BaseScreenState<SuccessMessageScreenWithBloc> baseScreenCreateState() =>
+      _SuccessMessageScreenWithBlocState();
+}
+
+class _SuccessMessageScreenWithBlocState
+    extends BaseScreenState<SuccessMessageScreenWithBloc> {
+  @override
+  Widget baseScreenBuild(BuildContext context) {
+    return BlocConsumer<SuccessMessageBloc, SuccessMessageState>(
+      listener: (context, state) {
+        if (state is DoneSuccessState) {
+          if (source == true) {
+            _openHomeScreen(context);
+          } else {
+            _openSetAccountScreen(context);
+          }
+        }
+      },
+      builder: (context, state) {
+        return AuthBase(
+          body: Container(
+            decoration: const BoxDecoration(
+              color: AppColors.backgroundOfWidget,
+              borderRadius: BorderRadius.only(
+                topRight: Radius.circular(10),
+                topLeft: Radius.circular(10),
+              ),
+            ),
+            child: Padding(
+              padding: const EdgeInsetsDirectional.only(
+                  start: 16, end: 16, top: 70, bottom: 24),
+              child: Column(
+                children: [
+                  SvgPicture.asset(AppAssetPaths.checkMarkSuccess),
+                  const SizedBox(
+                    height: 20,
+                  ),
+                  Text(
+                    translate(LocalizationKeys.congrats)!,
+                    style: const TextStyle(
+                      fontSize: 27,
+                      fontWeight: FontWeight.w700,
+                      color: AppColors.titleOfWidget,
+                    ),
+                  ),
+                  const SizedBox(
+                    height: 15,
+                  ),
+                  Text(
+                    source
+                        ? translate(LocalizationKeys.resetPasswordSuccessfully)!
+                        : translate(
+                            LocalizationKeys.youHaveSuccessfullyRegistered)!,
+                    style: const TextStyle(
+                      fontSize: 16,
+                      fontWeight: FontWeight.w500,
+                      color: AppColors.descriptionOfWidget,
+                    ),
+                  ),
+                  const SizedBox(
+                    height: 72,
+                  ),
+                  Row(
+                    children: [
+                      Expanded(
+                          child: AppButton(
+                        title: translate(LocalizationKeys.done)!,
+                        onPressed: () {
+                          BlocProvider.of<SuccessMessageBloc>(context)
+                              .add(DoneEvent());
+                        },
+                      )),
+                    ],
+                  ),
+                ],
+              ),
+            ),
+          ),
+        );
+      },
+    );
+  }
+
```

Comment:
pls, add helper methods and helper widgets comments in the whole app when needed